apps/projects/routes.py

```diff
@@ -1074,104 +1146,151 @@ def get(self, project_id):
                 'Accept-Ranges': 'bytes',
                 'Content-Length': chunksize,
-                'Content-Type': doc.get("mime_type"),
+                'Content-Type': self._project.get("mime_type"),
             }
             # get a stack of bytes push to client
-            stream = app.fs.get_range(doc['storage_id'], start, chunksize)
+            stream = app.fs.get_range(self._project['storage_id'], start, chunksize)
             res = make_response(stream)
             res.headers = headers
             return res, 206
 
         headers = {
             'Content-Length': length,
-            'Content-Type': 'video/mp4',
+            'Content-Type': self._project.get("mime_type"),
         }
-        stream = app.fs.get(doc.get('storage_id'))
+        stream = app.fs.get(self._project.get('storage_id'))
         res = make_response(stream)
         res.headers = headers
         return res, 200
 
 
 class GetRawThumbnail(MethodView):
-    SCHEME_THUMBNAIL = {
+    SCHEMA_THUMBNAILS = {
+        'type': {
+            'type': 'string',
+            'required': True,
+            'anyof': [
+                {
+                    'allowed': ['timeline'],
+                    'dependencies': ['index']
+                },
+                {
+                    'allowed': ['preview'],
+                    'excludes': 'index'
+                }
+            ],
+        },
         'index': {
             'type': 'integer',
-            'required': False,
-            'empty': True,
             'coerce': int,
-            'min': 0,
-        },	
+            'min': 0
+        }
     }
 
     def get(self, project_id):
         """
-        Get thumbnail
+        Get thumbnail file
         ---
         parameters:
         - in: path
           name: project_id
           type: string
           required: True
           description: Unique project id
-        - name: index
+        - name: type
           in: query
-          type:
-          - integer
-          - string
-          description: index of thumbnail to get, leave empty to get preview thumbnail
+          type: string
+          enum: [preview, timeline]
+        - in: query
+          name: index
+          type: integer
+          description: Index of timeline thumbnail to read. Used only when `type` is `timeline`.
         produces:
           - image/png
         responses:
           200:
-            description: OK
-            schema:
-              type: file
+            description: thumbnail image
+            content:
+              image/png:
+                schema:
+                  type: string
+                  format: binary
         """
-        doc = app.mongo.db.projects.find_one_or_404({'_id': format_id(project_id)})
 
-        if not request.args:
-            preview_thumbnail = doc.get('preview_thumbnail')
-            if not preview_thumbnail:
+        document = validate_document(request.args.to_dict(), self.SCHEMA_THUMBNAILS)
+
```

Comment:
I think we should set default value of `type` is preview thumbnail to avoid raising error when missing args.

Reply:
IMO it's ok to raise an error if we have a missing args for a request :) 
It's more explicit when `type` is required. 
I think we can do this:
- `projects/raw/thumbnails/preview` for preview thumb
- `projects/raw/thumbnails/timeline/<index>` for timeline thumb

What do you think?

Reply:
Yes, we agree with this.